src/main/java/dk/kb/image/Facade.java

```diff
@@ -0,0 +1,174 @@
+package dk.kb.image;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+
+import javax.imageio.ImageIO;
+import javax.swing.plaf.ColorUIResource;
+
+import com.ctc.wstx.cfg.OutputConfigFlags;
+
+import java.awt.image.BufferedImage;
+import java.awt.Color;
+
+public class Facade {
+    public static byte[] getGreyscale(BufferedImage img) throws IOException{
+        // get image's width and height
+        int width = img.getWidth();
+        int height = img.getHeight();
+  
+        // convert to grayscale
+        // Nested for loop
+        for (int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {               
+                // Here (x,y)denotes the coordinate of image
+                // for modifying the pixel value.
+                // Get RGB value of single pixel
+                int pixel = img.getRGB(x, y); 
+                // Shifts colors bitwise right
+                int a = (pixel >> 24) & 0xff;
+                int r = (pixel >> 16) & 0xff;
+                int g = (pixel >> 8) & 0xff;
+                int b = pixel & 0xff;
+                // calculate average
+                int avg = (r + g + b) / 3;
+                // replace RGB value with avg
+                pixel = (a << 24) | (avg << 16) | (avg << 8)
+                    | avg;
+                // set pixel to grayscale colors
+                img.setRGB(x, y, pixel);
+            }
+        }
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        // Write img to ByteArrayOutputStream as jpg
+        ImageIO.write(img, "jpg", baos);
+        return baos.toByteArray();
+    }
+
+    public static String getColorCount(BufferedImage img) throws IOException{
+        // Hashset is a collection that only has one of each value
+        Set<Integer> colors = new HashSet<Integer>();
+        // get image's width and height
+        int width = img.getWidth();
+        int height = img.getHeight();
+
+        // Nested for loop that adds unique colors to the hashset colors
+        for(int y = 0; y < height; y++) {
+            for(int x = 0; x < width; x++) {
+                int pixel = img.getRGB(x, y);     
+                colors.add(pixel);
+            }
+        }
+        String result = "There are " + Integer.toString(colors.size()) + " colors in this picture.";
+        return result;
+    }
+
+    public static String colorDistance(BufferedImage img){
+        // Define simple buckets
+        int[] buckets = defineSimpleBuckets();
+        // Count pixels and add to closest bucket
+        int[] bucketCount = countBucketsForImg(img, buckets);
+        // Get best bucket
+        int bestBucket = getBestBucket(bucketCount);
```

Comment:
Maybe "largest" or something like that would be better than "best"? "best is very subjective.